### PR TITLE
[stable/ark] CRD hooks and schedules

### LINKED ISSUE #6448: [stable/ark] optional backup schedule
FEATURE REQUEST:

I would like to automatically create an ark backup schedule when installing the chart. The schedule would not be created by default, but there would a Helm property to enable the creation of the schedule and properties to define the various schedule parameters.

### PR DESCRIPTION
**Pull request by plumdog**
**What this PR does / why we need it**:
- Refactors the Ark chart's handling of CRDs and creation of instances of those CRDs.
- Adds an option to add schedules via the helm chart.

**Which issue this PR fixes** : fixes #6448 #6438 

**Special notes for your reviewer**:
This makes use of helm's `crd-install` hook, which is included in helm 2.10, which has release candidates, but is not actually released yet. Using this means that we can get rid of the post-install hook that was running `kubectl create ...`, which is what caused the two issues linked above.

### CODE REVIEW

**Comment by paulczar** [comment]
I think we need to wait until helm 2.10 is released and we should probably document that it requires helm 2.10 or higher.

**Comment by plumdog** [comment]
@paulczar sounds good.

I will investigate whether the `pre-install` hook is a reasonable fallback.

**Comment by plumdog** [comment]
My experiments suggest that the answer to https://github.com/helm/charts/pull/7165#issuecomment-413026401 is "no".

I have set the required helm version in Chart.yaml.

This should not be considered for merging until helm 2.10 is released.

**Comment by plumdog** [comment]
Helm 2.10 is [now released](https://github.com/helm/helm/releases/tag/v2.10.0) which is lovely.

So I now claim that this PR should be considered for merging.

File: stable/ark/templates/config.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{{- $root := . }}
+{{ with .Values.configuration }}
```
Comment (unguiculus) [inline stable/ark/templates/config.yaml]: If you move this further down as it was, you don't need to capture the root context in a separate variable.

**Review by unguiculus — CHANGES_REQUESTED** [review]
I think this change warrants a major chart version bump.

**Comment by jrnt30** [comment]
This seems very aligned with the way I would want to approach this as well.  Started looking at upgrading to the 0.10 RC internally via Helm and this would make adjusting the CRDs much easier

**Comment by unguiculus** [comment]
FYI, there are issues with the new crd hook: https://github.com/helm/helm/issues/4697

**Comment by plumdog** [comment]
Have rebased and made the suggested CR change.

However the `crd-install` hook issues are a bit more worrying. Do we have an accepted _least bad_ way of handling CRD's in helm? Should the current state of `crd-install` be considered better or worse than the existing behaviour in this chart?

**Comment by unguiculus** [comment]
Sorry for the delay. Please fix DCO and resolve conflicts so we can move this forward.

**Comment by plumdog** [comment]
@unguiculus No problem!

Rebased and DCO-ed.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**